### PR TITLE
[Feature][Connector-V1 & V2] Support unauthorized ClickHouse

### PR DESCRIPTION
**Pull request by qianmoQ**
## Purpose of this pull request

Support ingesting data to ClickHouse without using account password...

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by Hisoka-X** [comment]
Can you add it to connector-v2?

**Comment by qianmoQ** [comment]
> Can you add it to connector-v2?

Ok, I'll fit it in

File: seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/sink/client/ClickhouseSink.java [hunk 1]
```diff
@@ -78,7 +78,14 @@ public String getPluginName() {
     @SuppressWarnings("checkstyle:MagicNumber")
     @Override
     public void prepare(Config config) throws PrepareFailException {
-        CheckResult result = CheckConfigUtil.checkAllExists(config, HOST, DATABASE, TABLE, USERNAME, PASSWORD);
+        CheckResult result = CheckConfigUtil.checkAllExists(config, HOST, DATABASE, TABLE);
+
+        boolean isCredential = config.hasPath(USERNAME) && config.hasPath(PASSWORD);
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/sink/client/ClickhouseSink.java]:
```suggestion
        boolean isCredential = config.hasPath(USERNAME) || config.hasPath(PASSWORD);
```

**Review by Hisoka-X — APPROVED** [review]
LGTM, thanks for your contribtue!

**Pull request merged**